### PR TITLE
Prepare for migration to Node 4.2 LTS

### PR DESCRIPTION
**Pull request by thisandagain**
- Updates `dependencies`
- Updates and removes redundant and otherwise unused modules from `devDependencies`. I attempted to keep this fairly minimal as to mitigate risk to our build process.
- Updates `travis.yml` to test against Node 4.2.X
- Add explicit support for `C++11` to resolve build issues in Travis CI `Precise` images

Resolves GH-351

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -52,10 +52,7 @@
     "minilog": "2.0.8",
     "node-sass": "3.3.3",
     "po2icu": "git://github.com/LLK/po2icu.git#develop",
-    "react": "0.14.0",
-    "react-addons-test-utils": "0.14.0",
-    "react-dom": "0.14.0",
-    "react-intl": "2.0.0-beta-1",
```
Comment (rschamp) [inline package.json]: I assume you removed these because we include them statically? Won't tests require them to run?

Comment (thisandagain) [inline package.json]: Correct. These aren't used in our current code as we include them in `./static/js/lib`.

**Comment by thisandagain** [comment]
Failing due to native dependency of `iconv` within `po2icu`:
`scratch-www/node_modules/po2icu/node_modules/i18next-conv/node_modules/node-gettext/node_modules/iconv/build`

**Comment by thisandagain** [comment]
Interestingly, I'm having trouble forcing Mac OS X to install `iconv` when running `npm install`. The `encoding` module *should* be using the non-native `iconv-lite` module by default.

**Comment by thisandagain** [comment]
The fact that `iconv` is being installed is one problem. The default travis compiler not supporting C++11 is another. Addressing this issue in 834327a. 

Reference:
https://docs.travis-ci.com/user/languages/javascript-with-nodejs#Node.js-v4-(or-io.js-v3)-compiler-requirements

**Pull request merged**